typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,42 @@
+const $tabs = document.querySelectorAll('.tab');
+if (!$tabs) throw new Error('.tab query failed!');
+
+const $tabContainer = document.querySelector('.tab-container');
+if (!$tabContainer) throw new Error('.tab-container query failed');
+
+const $views = document.querySelectorAll('.view');
+if (!$views) throw new Error('view query failed!');
+
+let currentTabNdx: number = 0;
+
+$tabContainer.addEventListener('click', (event: Event) => {
+  const eventTarget = event.target as HTMLElement;
+  console.log('eventTarget:', eventTarget);
+
+  if (!(eventTarget.className === 'tab')) {
+    console.log('not an inactive tab!');
+    return;
+  } // if what we clicked on was not a tab, return
+
+  $tabs[currentTabNdx].className = 'tab';
+  // previous current tab loses active class
+
+  for (let i = 0; i < $tabs.length; i++) {
+    if (event.target === $tabs[i]) {
+      $tabs[i].className = 'tab active';
```

Comment:
You should use your `eventTarget` variable here since you stored the `event.target` there and asserted its type to TypeScript